src_sycl/CMakeLists.txt

```diff
@@ -21,14 +21,43 @@ add_executable(nbodygl ${COMMON_SOURCE} ${OPENGL_SOURCE})
 
 target_compile_definitions(nbodygl PRIVATE -DUSE_OPENGL)
 
-target_compile_features(nbodygl PRIVATE cxx_auto_type cxx_nullptr cxx_range_for)
+target_compile_features(nbodygl PRIVATE cxx_auto_type cxx_nullptr cxx_range_for cxx_std_17)
 
 target_include_directories(nbodygl PRIVATE ${CMAKE_CURRENT_SOURCE_DIR}/include)
 
 target_link_libraries(nbodygl PRIVATE glm::glm glfw PkgConfig::Glew OpenGL::OpenGL)
 
-if(CMAKE_CXX_COMPILER MATCHES ".*dpcpp[.exe]*")
-	target_compile_options(nbodygl PRIVATE -fsycl)
-endif()
+option(USE_COMPUTECPP "whether to use ComputeCpp compiler" OFF)
+
+if(USE_COMPUTECPP)
+	find_package(ComputeCpp REQUIRED)
+
+	set(CMAKE_CXX_COMPILER ${ComputeCpp_DEVICE_COMPILER_EXECUTABLE})
+
+	set_property(SOURCE simulator.dp.cpp APPEND
+        PROPERTY COMPILE_OPTIONS
+		-sycl-driver -std=gnu++17 -O3
+		-mllvm -local_mem_pools=8
+		-mllvm -local_mem_pools_size=5000
+		-march=native -sycl-target host-x86_64)
```

Comment:
I don't know that any of these should be necessary. `-sycl-driver` can be implied, `-O3` is the default, new versions of FindComputeCpp.cmake will set `-std=xx` based on the `CXX_STANDARD` CMake property is, and the other output stuff is set by the `COMPUTECPP_BITCODE` options and so on.

If you don't use the sycl driver mode then you could set `-march=native` on the host compiler instead, to get some benefits to the generated host code.

Reply:
Found that, with the current setup, `-O3` does make a difference, but have removed the rest.